Source/driver/sdc_util.F90

```diff
@@ -42,19 +51,133 @@ module sdc_util
 
 #ifdef REACTIONS
   subroutine sdc_solve(dt_m, U_old, U_new, C, sdc_iteration)
-    ! the purpose of this function is to solve the system
-    ! U - dt R(U) = U_old + dt C
+    ! this is the main interface to solving the discretized nonlinear
+    ! reaction update.  It either directly calls the Newton method or first
+    ! tries VODE and then does the Newton update.
+
+    use meth_params_module, only : NVAR, UEDEN, UEINT, URHO, UFS, UMX, UMZ, UTEMP, &
+                                   sdc_solver
+    use amrex_constants_module, only : ZERO, HALF, ONE
+    use burn_type_module, only : burn_t
+    use react_util_module
+    use extern_probin_module, only : SMALL_X_SAFE
+    use network, only : nspec, nspec_evolve
+    use rpar_sdc_module
+
+    implicit none
+
+    real(rt), intent(in) :: dt_m
+    real(rt), intent(in) :: U_old(NVAR)
+    real(rt), intent(inout) :: U_new(NVAR)
+    real(rt), intent(in) :: C(NVAR)
+    integer, intent(in) :: sdc_iteration
+
+    integer :: ierr
+
+
+    if (sdc_solver == NEWTON_SOLVE) then
+       ! we are going to assume we already have a good guess for the
+       ! solving in U_new and just pass the solve onto the main Newton
+       ! solve
+       call sdc_newton_solve(dt_m, U_old, U_new, C, sdc_iteration, ierr)
```

Comment:
Similar to what we do for the CTU subcycling, it should be possible to have sdc_newton_subdivide be the main entry routine, which starts with nsub=1 and then takes more as needed. This makes the code slightly cleaner.

Reply:
yes, that makes sense.

Reply:
this has been fixed